tsdb/chunkenc/float_histogram.go

```diff
@@ -576,17 +581,17 @@ func (a *FloatHistogramAppender) AppendFloatHistogram(prev *FloatHistogramAppend
 			return nil, false, a, nil
 		}
 
-		if prev != nil && h.CounterResetHint != histogram.CounterReset {
+		if h.CounterResetHint == histogram.CounterReset {
+			// Always honor the explicit counter reset hint.
+			a.setCounterResetHeader(CounterReset)
+		} else if prev != nil {
```

Comment:
It's not a hard styleguide requirement, but I much prefer to use a switch statement in this case already (and that's what's mostly done throughout the codebase):

```suggestion
		switch {
		case h.CounterResetHint == histogram.CounterReset:
			// Always honor the explicit counter reset hint.
			a.setCounterResetHeader(CounterReset)
		case prev != nil:
```

It's about the same length, but the two conditions are presented in the same way (rather than one buried more deeply), and should you ever need to add a 3rd case, it's easier.